src/api/routes/test.test.ts

```diff
@@ -23,28 +27,113 @@ describe('test endpoints', () => {
-      );
+    it('returns 404 if user is not found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const unknowUserId = new UserId();
 
-      const test_saved = await testRepository.save(
-        new Test(new TestId(), user.id, new TestTypeId(), new Date())
-      );
+      await request(app)
+        .get(`/api/v1/users/${unknowUserId.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(404);
+    });
+
+    it('returns 200 with the existing test if user is found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const test = await testRepository.save(aTest(user.id));
 
       await request(app)
         .get(`/api/v1/users/${user.id.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(200)
+        .expect((response) => {
+          expect(response.body[0].id).toEqual(test.id.value);
+        });
+    });
+
+    it('returns 200 if a user with an access pass requests another users tests', async () => {
+      // TODO
+    });
+  });
+
+  describe('POST /users/:id/tests', () => {
+    it('returns 401 if user is not authenticated', async () => {
+      const userId = new UserId();
+      await request(app)
+        .post(`/api/v1/users/${userId.value}/tests`)
+        .expect(401);
+    });
+
+    it('returns 404 if user is not found', async () => {
+      const user = await userRepository.save(aNewUser());
+      const unknowUserId = new UserId();
+
+      await request(app)
+        .post(`/api/v1/users/${unknowUserId.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .expect(404);
+    });
+
+    it('returns 403 if a user with an access pass tries to create a test for another user', async () => {
+      // TODO
+    });
+
+    it('returns 422 if the test type does not exist', async () => {
+      const user = await userRepository.save(aNewUser());
+      const testType = aTestType();
+
+      await request(app)
+        .post(`/api/v1/users/${user.id.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .send({ testTypeId: testType.id.value })
+        .expect(422);
+    });
+
+    it('returns 422 if there are malformed results', async () => {
+      const user = await userRepository.save(aNewUser());
+      const testType = await testTypeRepository.save(aTestType());
+
+      await request(app)
+        .post(`/api/v1/users/${user.id.value}/tests`)
+        .set({ Authorization: `Bearer ${await getTokenForUser(user)}` })
+        .send({
+          testTypeId: testType.id.value,
+          results: {
+            testerUserId: user.id.value,
+            details: { c: 12, igg: 'Wot', igm: [] },
+          },
+        })
+        .expect(422);
+    });
+
+    it('returns 200 with the existing test if user is found', async () => {
```

Comment:
Couple of comments on this:
1. I would expect POST /resource to return 201 Created on success
2. Given this is a new test the test name should say something along the lines of  `returns 201 with the newly created test if user is found`